### PR TITLE
improvement: clarify some confusion around --all, --force and --scope flags of "bit tag".

### PR DESCRIPTION
**Pull request by davidfirst**
## Proposed Changes

- deprecate `--scope`. replaced by `--workspace`. 
- disallow using `--force` and `--all` (before, in this case "--force" did nothing. now it's disabled to clear confusions).
- update docs of `--force` and `--scope/workspace`.

Asked by @benjgil .

### CODE REVIEW

File: e2e/commands/snap.e2e.2.ts [hunk 1]
```diff
@@ -452,7 +452,7 @@ describe('bit snap command', function () {
             expect(output).to.have.string('unable to snap/tag "bar/foo", it is unmerged with conflicts');
           });
           it('should not include the component when running bit tag --all', () => {
-            const output = helper.general.runWithTryCatch('bit tag -a -f');
```
Comment (benjgil) [inline e2e/commands/snap.e2e.2.ts:455]: add a test for the output when -a and -f are used together?

File: scopes/workspace/workspace/tag-cmd.ts [hunk 2]
```diff
@@ -38,7 +39,7 @@ ${WILDCARD_HELP('tag')}`;
     ['', 'minor', 'increment the minor version number'],
     ['', 'major', 'increment the major version number'],
     ['', 'pre-release [identifier]', 'EXPERIMENTAL. increment a pre-release version (e.g. 1.0.0-dev.1)'],
-    ['f', 'force', 'force-tag even if tests are failing and even when component has not changed'],
+    ['f', 'force', 'force-tag even when the component has not changed. this cannot be used with "--all"'],
```
Comment (benjgil) [inline scopes/workspace/workspace/tag-cmd.ts:42]: 1. Would remove the this, so "...changed. Cannot be used with "--all""

File: src/api/consumer/lib/tag.ts [hunk 3]
```diff
@@ -51,26 +52,31 @@ type TagParams = {
...
-  const isAll = Boolean(all || scope || idsHasWildcard);
   const validExactVersion = validateVersion(exactVersion);
-  const preHook = isAll ? PRE_TAG_ALL_HOOK : PRE_TAG_HOOK;
+  const isMany = Boolean(all || workspace || idsHasWildcard);
+  const preHook = isMany ? PRE_TAG_ALL_HOOK : PRE_TAG_HOOK;
   HooksManagerInstance.triggerHook(preHook, tagParams);
   const consumer = await loadConsumer();
+  if (!consumer.isLegacy) {
+    if (force && all) {
+      throw new GeneralError('you can use either "--force" or "--all", but not both. consider using "--workspace"');
```
Comment (benjgil) [inline src/api/consumer/lib/tag.ts:71]: error text suggestion (less confusing):
'"--all" cannot be used with "--force". To force-tag all components in workspace use "--force --workspace"'

**Comment by davidfirst** [comment]
After some more discussion with @itaymendel and @benjgil , the following changes will be done:
1. deprecate the `--all` flag. This will be the default - tagging all new/modified components. It'll make it consistent over the board with other commands, such as `build`, `compile`, `test`. 
2. replace `--scope` flag with `--unmodified`. Now that the default is to tag new & modified, this flag is clearer about including the unmodified components.
3. the `--force` flag will be deprecated. It has no use anymore. To skip tests, use `--skip-tests`, to include unmodified, use `--unmodified`. 

(The same will apply to `bit snap` as well)

@ranm8 , @GiladShoham , let me know if you have rejections.

**Comment by GiladShoham** [comment]
@davidfirst what if I do bit tag comp-id and that component is not modified?
Should I use --unmodified?

**Comment by davidfirst** [comment]
@GiladShoham , yes. You should run `bit tag comp1 --unmodified` in this case.

**Comment by davidfirst** [comment]
This PR is not relevant anymore as it we took a different approach, See https://github.com/teambit/bit/pull/5692 for the details.

**Pull request closed without merging**